addons/html_builder/static/src/builder/options/device_visibility_option.js

```diff
@@ -0,0 +1,97 @@
+import { registry } from "@web/core/registry";
+import { Plugin } from "@html_editor/plugin";
+class DeviceVisibilityOptionPlugin extends Plugin {
+    static id = "DeviceVisibilityOption";
+    websiteService = this.services.website;
+    selector = "section .row > div";
+    resources = {
+        builder_options: {
+            template: "html_builder.DeviceVisibilityOption",
+            selector: this.selector,
+        },
+        builder_actions: this.getActions(),
+        clean_for_save_handlers: this.cleanForSave.bind(this),
+    };
+    cleanForSave({ root }) {
+        const editingEls = root.querySelectorAll(this.selector);
+        for (const editingEl of editingEls) {
+            const invisibleClass = editingEl.classList.contains("o_snippet_desktop_invisible")
+                ? "no_desktop"
+                : "" || editingEl.classList.contains("o_snippet_desktop_invisible")
+                ? "no_mobile"
+                : "";
+            if (invisibleClass) {
+                this.getActions().toggleDeviceVisibility.apply({
+                    editingElement: editingEl,
+                    param: invisibleClass,
+                });
+                editingEl.classList.remove("o_snippet_override_invisible");
+            }
```

Comment:
Question: Why do you need to do all that and apply the option again (toggleDeviceVisibility) ? 
The master cleanforsave was just removing the `'o_snippet_override_invisible'` class, because it is only needed in edit mode, so you can always remove it safely :slightly_smiling_face: